src/pulp/solvers.py

```diff
@@ -2887,3 +2887,242 @@ def __init__(self, path=None, *args, **kwargs):
+                                        mosek.solsta.unknown: LpStatusUndefined,
+                                        mosek.solsta.integer_optimal: LpStatusOptimal,
+                                        mosek.solsta.prim_illposed_cer:LpStatusNotSolved, 
+                                        mosek.solsta.dual_illposed_cer: LpStatusNotSolved,
+                                        mosek.solsta.prim_feas: LpStatusNotSolved,
+                                        mosek.solsta.dual_feas: LpStatusNotSolved,
+                                        mosek.solsta.prim_and_dual_feas: LpStatusNotSolved}
+            #Variable values.
+            try:
+                self.xx = [0.]*self.numvars
+                self.task.getxx(self.solution_type,self.xx)
+                for var in lp.variables():
+                    var.varValue = self.xx[self.var_dict[var.name]]
+            except mosek.Error:
+                pass
+            #Constraint slack variables.
+            try:
+                self.xc = [0.]*self.numcons
+                self.task.getxc(self.solution_type,self.xc)
+                for con in lp.constraints:
+                    lp.constraints[con].slack = -(self.cons[con].constant + self.xc[self.cons_dict[con]])
+            except mosek.Error:
+                pass
+            #Reduced costs.
+            if self.solution_type != mosek.soltype.itg:
+                try:
+                    self.x_rc = [0.]*self.numvars
+                    self.task.getreducedcosts(self.solution_type,0,self.numvars,self.x_rc)
+                    for var in lp.variables():
+                        var.dj = self.x_rc[self.var_dict[var.name]]
+                except mosek.Error:
+                    pass
+            #Constraint Pi variables.
+                try:
+                    self.y = [0.]*self.numcons
+                    self.task.gety(self.solution_type,self.y)
+                    for con in lp.constraints:
+                        lp.constraints[con].pi = self.y[self.cons_dict[con]]
+                except mosek.Error:
+                    pass
+            
+        def putparam(self, par, val):
+            if isinstance(par, mosek.dparam):
+                self.task.putdouparam(par, val)
+            elif isinstance(par, mosek.iparam):
+                self.task.putintparam(par, val)
+            elif isinstance(par, mosek.sparam):
+                self.task.putstrparam(par, val)
+            elif isinstance(par, str):
+                if par.startswith("MSK_DPAR_"):
+                    self.task.putnadouparam(par, val)
+                elif par.startswith("MSK_IPAR_"):
+                    self.task.putnaintparam(par, val)
+                elif par.startswith("MSK_SPAR_"):
+                    self.task.putnastrparam(par, val)
+                else:
+                    raise Exception("Invalid MOSEK parameter '%s'." % par)
+
+        def actualSolve(self, lp):
+            self.buildSolverModel(lp)
```

Comment:
After making the `lp` problem using pulp, the user could initialize the Mosek solver by passing a dictionary of any and all parameters that Mosek supports (for example: `lp.setSolver(solver=MOSEK(mosek_pars={'valid_par':value}))`) and then proceed to solve the problem with `actualSolve.` 

If by your third point you meant to suggest that the user should be able to manipulate the _Task_ object itself (within MOSEK's Optimizer API), then I could include a method that returns the Mosek _Task_ object. However, I am unsure about how helpful this would really be, especially over the parameter-dictionary mentioned above.